src/nxdoc/nuxeo-server/upgrading-the-nuxeo-platform/how-to-upgrade-from-lts-2023-to-lts-2025/how-to-upgrade-search-service.md

```diff
@@ -0,0 +1,301 @@
+SearchResponse response = esa.getClient().search(request);
+```
+
+This requires to add a dependency on `nuxeo-runtime-opensearch1` module, to be converted to:
+```java
+import org.nuxeo.runtime.opensearch1.OpenSearchClientService;
+import org.nuxeo.runtime.opensearch1.client.OpenSearchClient;
+
+OpenSearchClient client = Framework.getService(OpenSearchClientService.class).getClient("default");
+SearchRequest request = new SearchRequest(IDX_NAME)
+  .source(new SearchSourceBuilder().query(QueryBuilders.matchAllQuery()).from(0).size(60));
+SearchResponse response = client.search(request);
+```
+
+### Test runner
+
+There is no more `RepositoryElasticSearchFeature` or `RepositoryLightElasticSearchFeature` feature runners.
+
+You have to use the new `CoreSearchFeature`.
+
+```java
+@RunWith(FeaturesRunner.class)
+@Features(CoreSearchFeature.class)
+public class MyTestSuite {
+
+  @Inject
+  protected CoreSession session;
+
+  @Inject
+  protected SearchService searchService;
+
+  @Inject
+  protected TransactionalFeature txFeature;
+
+  @Test
+  public void testSomething() {
+    ...
+    var documents = searchService.search(SearchQuery.builder("SELECT * FROM Document).build).loadDocuments(session);
+    ... 
+  }
+}
+
+```
+It's possible to run the test suite with different search clients:
+```bash
+# run test with default repository search client
+mvn -nsu test -Dtest=MyTestSuite
+# run unit test against an embedded opensearch1 server
+mvn -nsu test -Dtest=MyTestSuite -Dnuxeo.test.search=opensearch1
+# run unit test against a remote opensearch server
+mvn -nsu test -Dtest=MyTestSuite -Dnuxeo.test.search=opensearch1 -Dnuxeo.test.opensearch1.servers=http://my-opensearch-cluster:9200/
+```
+
+## Page Providers
+
+Any reference to `ElasticSearchNxqlPageProvider` page provider needs to be replaced by  `SearchServicePageProvider`.
+
+When an enhanced search client like `nuxeo-search-client-opensearch1` is installed it will be used.
+
+For instance, this
+```xml
+<genericPageProvider class="org.nuxeo.elasticsearch.provider.ElasticSearchNxqlPageProvider"
+```
+Should be replaced by:
+```xml
+<genericPageProvider class="org.nuxeo.ecm.platform.query.nxql.SearchServicePageProvider"
+```
+
+## XML Contributions
+Contributions made to `org.nuxeo.elasticsearch.ElasticSearchComponent` extension points should be updated to the new services. 
+
+For instance `hint` contributions should be changed from
+```xml
+  <extension target="org.nuxeo.elasticsearch.ElasticSearchComponent" point="elasticSearchHints">
+    <hint name="foo" class="..." />
+```
+to 
+```xml
+<extension target="org.nuxeo.ecm.core.search.client.opensearch1" point="hint">
+  <hint name="foo" class="..." />
+```
+
+## Automation Operations
+
+`ElasticSearch.Index` should be replaced by `Search.Index`.
+
+`Elasticsearch.BulkIndex` should be replaced by `Search.Index`.
+
+`Elasticsearch.WaitForIndexing` should be replaced by `Search.WaitForIndexing`.
```

Comment:
Maybe we should talk about operation parameters change?

Reply:
I have created https://hyland.atlassian.net/browse/NXP-33161 to accept the document model.